v1/nix/modules/drv-parts/fetch-pip/interface.nix

```diff
@@ -0,0 +1,85 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  l = lib // builtins;
+  t = l.types;
+in {
+  options.fetch-pip = {
+    hash = l.mkOption {
+      type = t.str;
+      description = ''
+        hash for the fixed output derivation
+      '';
+    };
+    maxDate = l.mkOption {
+      type = t.str;
+      description = ''
+        maximum release date for packages
+        Choose any date from the past.
+      '';
+      example = "2023-01-01";
```

Comment:
The purpose of maxDate is not to signal when the FOD was last updated. It specifies the date for the snapshot view on pypi. Maybe we should rename it to `pypiSnapshotDate` or something like that.

As this is not an automatically managed filed but one that faces the user and should be set and modified by the user, I don't think we should make this a timestamp by default. It reduces the readability but doesn't bring us much benefit as long as there is a clear one to one mapping between date and timestamp.

Reply:
done the renaming in 56d9e582bdc2a6ddfb959758f495a484caa56d96

Reply:
`The purpose of maxDate is not to signal when the FOD was last updated. It specifies the date for the snapshot view on pypi. Maybe we should rename it to pypiSnapshotDate or something like that.`

By "updated" i mean the lock in this case, not every rebuild of a FOD. What's the benefit of keeping this in source after we moved to manage the hash automatically?

> I don't think we should make this a timestamp by default. It reduces the readability but doesn't bring us much benefit as long as there is a clear one to one mapping between date and timestamp.

You are right, we implicitly convert it to a timestamp anyways, just the naming was a bit confusing.

Reply:
> What's the benefit of keeping this in source after we moved to manage the hash automatically?

Good question. Since the pypiSnapshotDate effectively pins all the package versions, I don't feel that it should be updated on arbitrary changes. For example, you might want to add a new dependency, without updating all other dependencies.

Generally it seems like we'd need a feature that allows the user to only update individual values of the lock files. But maybe we can get around this for now by just keeping that value in the source.